### PR TITLE
Feature/create if necessary

### PR DESCRIPTION
**Pull request by HannesSommer**

### CODE REVIEW

File: sm_property_tree/src/BoostPropertyTree.cpp [hunk 1]
```diff
@@ -76,9 +76,8 @@ namespace sm {
     return dynamic_cast<const BoostPropertyTreeImplementation*>(_imp.get())->end();
   }
 
-  void BoostPropertyTree::update(const BoostPropertyTree & with)
+  void BoostPropertyTree::update(const BoostPropertyTree & with, bool createIfNecessary)
   {
-    return dynamic_cast< BoostPropertyTreeImplementation*>(_imp.get())->update(*dynamic_cast<const BoostPropertyTreeImplementation*>(with._imp.get()));
+    return dynamic_cast< BoostPropertyTreeImplementation*>(_imp.get())->update(*dynamic_cast<const BoostPropertyTreeImplementation*>(with._imp.get()), createIfNecessary);
```
Comment (simonlynen) [inline sm_property_tree/src/BoostPropertyTree.cpp]: 80/100c

File: sm_property_tree/src/BoostPropertyTreeImplementation.cpp [hunk 2]
```diff
@@ -180,6 +180,13 @@ namespace sm {
     }
   }
 
+  void updateOnly(boost::property_tree::ptree &dest, const boost::property_tree::ptree::path_type &childPath, const boost::property_tree::ptree &child) {
```
Comment (simonlynen) [inline sm_property_tree/src/BoostPropertyTreeImplementation.cpp]: 80/100c

File: sm_property_tree/include/sm/BoostPropertyTree.hpp [hunk 3]
```diff
@@ -45,9 +45,11 @@ namespace sm {
 
     /**
      * Update this ptree with another, allowing merging of ptrees
-     * @param with
+     * @param with the other ptree
+     * @param createIfNecessary it true (default) all necessary trees will be created automatically to store the values of with.
+     * If not it will throw an KeyNotFoundException if there is no such key already
      */
-    void update(const BoostPropertyTree & with);
+    void update(const BoostPropertyTree & with, bool createIfNecessary = true);
```
Comment (simonlynen) [inline sm_property_tree/include/sm/BoostPropertyTree.hpp]: default values are discouraged. Can you rather make an overload and call through with default = true.

**Comment by HannesSommer** [comment]
Thanks. Addressed. Avoiding the default instantaneously revealed an untreated interface change in sm_python :).

**Pull request merged**